flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/WithinCheckpointFileMergingSnapshotManager.java

```diff
@@ -41,12 +41,13 @@ public class WithinCheckpointFileMergingSnapshotManager extends FileMergingSnaps
      * this map consist of checkpoint id, subtask key, and checkpoint scope, which collectively
      * determine the physical file to be reused.
      */
-    private final Map<Tuple3<Long, SubtaskKey, CheckpointedStateScope>, PhysicalFile>
+    private final Map<Tuple3<Long, SubtaskKey, CheckpointedStateScope>, PhysicalFilePool>
```

Comment:
I'm a little bit lost here. Does a pool should contain and manage a set of physical files that may resue multiple times? Considering the files may share across SubtaskKeys (In exclusive scope), I suggest we leave ` Map<Long, PhysicalFilePool>` here, and let PhysicalFilePool handle the file sharing between subtasks in Exclusive scope and exclusive between Shared Scope.

Reply:
Thanks for the reminder. I have updated to share files across subtaskkeys in exclusive scope.